code/script.js

```diff
@@ -218,108 +220,141 @@ const generateBoard = () => {
+// It´ll filter the characters array and redraw the game board.
+// Alert popup that tells if´s: yes - else: no
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+  
+    if (category === 'accessories') {
+      if (keep) {
+        alert(`Yessss, that person wears ${value}! save all that wears ${value}`);
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
+      } else {
+        alert(`Nooo, that person doesn't wear ${value}! Remove all people who wears ${value}`);
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
+      }
+   }
+    else if (category === 'other') {
+      if (keep) {
+        alert(`Yeeeep! That´s right, the person is a ${value}! Keep all ${value}s here!`);
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
+      } else {
+        alert(`Nope! The person is not a ${value}! Remove all of the ${value}s`);
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
+      }
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
 
+    else {
+      if (keep) {
+        alert(`Yes, the person have ${value} ${category}! Keep all people that have ${value} ${category}`);
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
+      } else {
+        alert(`Noo! The person do not not have ${value} ${category}. Remove all the ones with ${value} ${category}`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
+      }
+    }
+   
   // Determine what is the category
   // filter by category to keep or remove based on the keep variable.
   /* 
     for hair and eyes :
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
       or
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
     for accessories and other
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
       or
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
   */
-
-  // Invoke a function to redraw the board with the remaining people.
+  //generetes the board 
```

Comment:
These small changes and add-ons to the pre-existing comments that you write throughout your document are great, they help clarifying the instructions and make sense of the changes you've made.